File: android/DFM/src/main/java/com/dontflymoney/entities/Nature.java
package com.dontflymoney.entities;

public enum Nature
{
    Out,
    In,
    Transfer;

    public int GetNumber()
    {
        switch (this)
        {
            case Out: return 0;
            case In: return 1;
            case Transfer: return 2;
            default: throw new UnsupportedOperationException();
        }
    }

    public static Nature GetNature(int number)
    {
        switch (number)
        {
            case 0: return Out;
            case 1: return In;
            case 2: return Transfer;
            default: throw new UnsupportedOperationException();
        }
    }
}

File: android/DFM/src/main/kotlin/com.darakeon.dfm/api/old/entities/Nature.kt
package com.darakeon.dfm.api.old.entities

enum class Nature {
	Out,
	In,
	Transfer;

	fun getNumber(): Int =
		when (this) {
			Out -> 0
			In -> 1
			Transfer -> 2
		}

	companion object {

		fun getNature(number: Int?): Nature =
			when (number) {
				0 -> Out
				1 -> In
				2 -> Transfer
				else -> throw UnsupportedOperationException()
			}
	}
}
